cmd/cmd.go

```diff
@@ -53,6 +55,14 @@ func ParseCLI() (err error) {
 
 	go interruptWatcher()
 
+	// TODO: too janky, need to do a better job here
+	if !strings.Contains(os.Args[0], "choria") {
+		builder := appbuilder.NewAppBuilder(ctx, filepath.Base(os.Args[0]))
+		if builder.HasDefinition() {
+			builder.RunCommand(ctx)
```

Comment:
I think we want to exit when we have run an "app"?
```suggestion
			builder.RunCommand(ctx)
			return
```

Reply:
Yeah actually fixed that in a uncommitted change here already. 

I am sure there are many horrors here hehe

Reply:
A `return` cause more trouble than it is supposed to fix, I could workaround the issue with this:
```diff
diff --git a/cmd/cmd.go b/cmd/cmd.go
index 7ddcdca8..4774787a 100644
--- a/cmd/cmd.go
+++ b/cmd/cmd.go
@@ -50,7 +50,7 @@ var (
 	ran        bool
 )
 
-func ParseCLI() (err error) {
+func ParseCLI() (wantsToRun bool, err error) {
 	ctx, cancel = context.WithCancel(context.Background())
 
 	go interruptWatcher()
@@ -60,6 +60,7 @@ func ParseCLI() (err error) {
 		builder := appbuilder.NewAppBuilder(ctx, filepath.Base(os.Args[0]))
 		if builder.HasDefinition() {
 			builder.RunCommand(ctx)
+			return false, nil
 		}
 	}
 
@@ -81,12 +82,12 @@ func ParseCLI() (err error) {
 		if cmd.FullCommand() == cli.command {
 			err = cmd.Configure()
 			if err != nil {
-				return fmt.Errorf("%s failed to configure: %s", cmd.FullCommand(), err)
+				return false, fmt.Errorf("%s failed to configure: %s", cmd.FullCommand(), err)
 			}
 		}
 	}
 
-	return
+	return true, nil
 }
 
 func systemConfigureIfRoot(actAsServer bool) error {
diff --git a/main.go b/main.go
index e2234ffb..05f16a1d 100644
--- a/main.go
+++ b/main.go
@@ -14,16 +14,19 @@ import (
 
 func main() {
 	var err error
+	var wantsToRun bool
 
-	err = cmd.ParseCLI()
+	wantsToRun, err = cmd.ParseCLI()
 	if err != nil {
 		log.Fatalf("Could not configure Choria: %s", err.Error())
 		os.Exit(1)
 	}
 
-	err = cmd.Run()
-	if err != nil {
-		log.Fatalf("Could not run Choria: %s", err.Error())
-		os.Exit(1)
+	if wantsToRun {
+		err = cmd.Run()
+		if err != nil {
+			log.Fatalf("Could not run Choria: %s", err.Error())
+			os.Exit(1)
+		}
 	}
 }
```

The comment in ParseCLI makes me think you want to do some refactoring here, just adding this to be sure it does not slip through :wink:

Reply:
did something simple - just had this before your suggestion - think its ok, but as you say, I have some itterations left to do here